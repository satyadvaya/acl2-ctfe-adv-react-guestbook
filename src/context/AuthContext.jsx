import React, { createContext, useContext, useState } from 'react';

const AuthContext = createContext();

const AuthProvider = ({ children }) => {
  const [user, setUser] = useState('');

  const login = (username, password) => {
    const loginMatch =
      username === process.env.AUTH_USERNAME &&
      password === process.env.AUTH_PASSWORD;
    if (loginMatch) setUser({ username });
    return loginMatch;
  };

  const logout = () => {
    setUser(null);
  };

  return (
    <AuthContext.Provider value={{ user, login, logout }}>
      {children}
    </AuthContext.Provider>
  );
};

const useAuth = () => {
  const context = useContext(AuthContext);
  if (context === undefined) {
    throw new Error(
      'custom hook useAuth must be used within an AuthContext Provider'
    );
  }

  return context;
};

const useForm = (inputs = {}) => {
  const [formState, setFormState] = useState(inputs);

  const handleFormChange = (event) => {
    const { name, value } = event.target;
    setFormState((prevState) => {
      return { ...prevState, [name]: value };
    });
  };
  return { formState, handleFormChange };
};

export { AuthProvider, useAuth, useForm };
