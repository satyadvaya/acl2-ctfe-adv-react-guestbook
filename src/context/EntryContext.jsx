import React, { createContext, useContext, useState } from 'react';

const EntryContext = createContext();
const EntryProvider = ({ children }) => {
  const [entries, setEntries] = useState([]);

  return (
    <EntryContext.Provider value={{ entries, setEntries }}>
      {children}
    </EntryContext.Provider>
  );
};

const useEntries = () => {
  const context = useContext(EntryContext);

  if (context === undefined) {
    throw new Error(
      'custom hook useEntries must be used within an EntryContext Provider'
    );
  }

  return context;
};

export { EntryProvider, useEntries };
