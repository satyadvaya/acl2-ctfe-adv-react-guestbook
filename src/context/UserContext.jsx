import { createContext, useContext, useState } from 'react';

const UserContext = createContext();

const UserProvider = ({ children }) => {
  const [user, setUser] = useState('');

  return (
    <UserContext.Provider value={{ user, setUser }}>
      {children}
    </UserContext.Provider>
  );
};

const useUser = () => {
  const context = useContext(UserContext);

  if (context === undefined) {
    throw new Error(
      'custom hook useUser must be used within a UserContext Provider'
    );
  }

  return context;
};

export { UserProvider, useUser };
