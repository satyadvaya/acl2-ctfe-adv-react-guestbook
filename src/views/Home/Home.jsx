import React from 'react';
import GuestBook from '../../components/GuestBook/GuestBook';
import EntryList from '../../components/EntryList/EntryList';

export default function Home() {
  return (
    <div>
      <GuestBook />
      <EntryList />
    </div>
  );
}
