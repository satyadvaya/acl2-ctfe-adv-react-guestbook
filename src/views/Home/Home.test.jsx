import React from 'react';
import { render } from '@testing-library/react';
import { EntryProvider } from '../../context/EntryContext';
import { UserProvider } from '../../context/UserContext';
import Home from './Home';

it('should render the Home component', () => {
  const { container } = render(
    <EntryProvider>
      <UserProvider>
        <Home />
      </UserProvider>
    </EntryProvider>
  );

  expect(container).toMatchSnapshot();
});
