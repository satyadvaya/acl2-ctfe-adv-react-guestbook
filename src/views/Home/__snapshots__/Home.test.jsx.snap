// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the Home component 1`] = `
<div>
  <div>
    <h1>
      Please Sign the Guestbook!
    </h1>
    <form>
      <div>
        <label
          for="guestName"
        >
          Guest Name
        </label>
        <input
          id="guestName"
          placeholder="Guest Name..."
          type="text"
          value=""
        />
      </div>
      <label
        for="guestEntry"
      >
        Guest Entry
      </label>
      <textarea
        id="guestEntry"
        placeholder="Your Entry!"
      />
      <button
        type="submit"
      >
        Sign üñäÔ∏è
      </button>
      
    </form>
    <ul />
  </div>
</div>
`;
